### ISSUE TITLE
duckdb: Trailing Commas in Values Clause

### ISSUE THREAD

**Issue by paultiq** [sqlfluff/sqlfluff#7451]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Trailing commas in a values clause fails to parse:
```sql
    SELECT *,
    FROM (VALUES ('Amsterdam', 1),
    ('London', 2),
    ) cities(name, id);
```

https://duckdb.org/docs/stable/sql/dialect/friendly_sql#trailing-commas

This query executes successfully in duckdb.

### Expected Behaviour

Should parse successfully, as this is valid duckdb

### Observed Behaviour

L:   2 | P:   5 |  PRS | Line 2, Position 5: Found unparsable section: "FROM
                       | (VALUES ('Amsterdam', 1),\n    ('Lon..."

### How to reproduce

sqlfluff lint test.sql

### Dialect

duckdb

### Version

4.0.0 and main

### Configuration

[tool.sqlfluff.core]
dialect = "duckdb"
templater = "jinja"

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
